### PR TITLE
feat: redux implementation

### PR DESCRIPTION
**Pull request by victorwu**
resolves FabricLabs/portal#393

### CODE REVIEW

File: src/Home.js [hunk 1]
```diff
@@ -28,39 +34,30 @@ import FaucetDripForm from './components/FaucetDripForm';
 // import FabricIdentityManager from './components/FabricIdentityManager';
 // import PortalMenu from './components/PortalMenu';
 
-class Home extends FabricComponent {
-  constructor (props) {
-    super(props);
-
-    this.settings = Object.assign({
+export default function Home (props) {
```
Comment (martindale) [inline src/Home.js]: Would love to preserve use of classes if possible

File: src/app/store.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { configureStore } from '@reduxjs/toolkit'
+import faucetReducer from '../features/faucet/faucetSlice'
+
+export const store = configureStore({
+  reducer: faucetReducer,
+})
```
Comment (martindale) [inline src/app/store.js]: Missing newline + semicolon

File: src/features/faucet/faucetSlice.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+})
+
+export const { 
+    recipientAddressClear, 
+    recipientAddressUpdate, 
+    statusUpdate, 
+    setState 
+} = faucetSlice.actions
+
+export default faucetSlice.reducer
```
Comment (martindale) [inline src/features/faucet/faucetSlice.js]: 4 spaces indentation, should be 2

**Review by martindale — CHANGES_REQUESTED** [review]
Minor nits, otherwise looks good

File: src/Home.js (hunk 1 above)
Comment (victorwu) [reply src/Home.js]: Updated via 5eaeacfb63ad952aa0ccd5b5e1793ae1fd184c7b && cfcc31e3e6439e8e0d99ca371cba8d05e2dacc8a

File: src/app/store.js (hunk 2 above)
Comment (victorwu) [reply src/app/store.js]: updated via fdfa0ceb8e84fe01184b778e145f0200994c631d

File: src/features/faucet/faucetSlice.js (hunk 3 above)
Comment (victorwu) [reply src/features/faucet/faucetSlice.js]: updated via fdfa0ceb8e84fe01184b778e145f0200994c631d

Comment (martindale) [reply src/features/faucet/faucetSlice.js]: Awesome, thanks!

**Review by martindale — APPROVED** [review]
Awesome work!  LGTM 🚢

**Pull request merged**